pulsar-metadata/src/main/java/org/apache/pulsar/metadata/bookkeeper/PulsarLedgerUnderreplicationManager.java

```diff
@@ -105,14 +105,17 @@ long getLedgerNodeVersion() {
     private final String urLockPath;
     private final String layoutPath;
     private final String lostBookieRecoveryDelayPath;
+    private final String replicationDisablePath;
     private final String checkAllLedgersCtimePath;
     private final String placementPolicyCheckCtimePath;
     private final String replicasCheckCtimePath;
 
     private final MetadataStoreExtended store;
 
-    private BookkeeperInternalCallbacks.GenericCallback<Void> replicationEnabledListener;
-    private BookkeeperInternalCallbacks.GenericCallback<Void> lostBookieRecoveryDelayListener;
```

Comment:
Why change the type to a collection? Only the latest event needs to be notified, right?

Reply:
Please review https://github.com/apache/pulsar/pull/21161 first.

Reply:
> Why change the type to a collection? Only the latest event needs to be notified, right?

Not, if the user registers two callbacks, the previous callback shouldn't be overridden. It also need trigger callback.